File: Java/Messager/app/src/main/java/bjfu/it/zhangsixuan/messager/ReceiveMessageActivity.java
package bjfu.it.zhangsixuan.messager;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class ReceiveMessageActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_receive_message);

        //get message
        Intent intent = getIntent();
        String message = intent.getStringExtra(CreateMessageActivty.MESSAGE_KEY);

        //set message
        TextView textView = findViewById(R.id.output);
        textView.setText(message);
    }
}

File: Kotlin/Messager/app/src/main/java/bjfu/it/zhangsixuan/messager/ReceiveMessageActivity.kt
package bjfu.it.zhangsixuan.messager

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_receive_message.*

class ReceiveMessageActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_receive_message)
        output.text =intent.getStringExtra(CreateMessageActivity.MESSAGE_KEY)

    }
}
